### DISCUSSION TITLE
HTTP Control Server: Status types and meanings

### DISCUSSION THREAD

**Discussion by yesimanoob** [qdm12/gluetun#653] [Q&A] [answered]
Hello! Can you please enumerate the different status types returned by the control server and their meanings? They don't seem to be documented here: [HTTP Control Server](https://github.com/qdm12/gluetun/wiki/HTTP-control-server)

Background:
I've got a few containers I'd like to join to gluetun's network and I want to add a startup script in each that waits for VPN connection in gluetun to complete before proceeding with the rest of the container launch process. I do this because my other containers launch pretty quickly, often more quickly than the vpn container they depend on can setup outbound firewalls to prevent any traffic from leaking before the tunnel is ready.

I'm looking for a hook into gluetun to get similar information. As such I'm hoping that there is a status that clarifies vpn connection status.

A different vpn container I use includes a proxy server that launches after the vpn connection completes (and firewall is running), and in that case I can simply wait for 'nc -z address port' to return success but it's looking like that may not work with gluetun.

**Comment by qdm12** [comment, marked as answer]
> can setup outbound firewalls to prevent any traffic from leaking before the tunnel is ready.

No need, gluetun blocks all traffic from the start. Well to be clear, for now it takes 60ms for the firewall to be fully up. Together with the overhead of Docker waiting for the container to be created to attach other containers to its networking stack, I think nothing would go through. I'm working now on decreasing that 60ms to 6ms, I'll let you know once I'm done.

Anyway, if you really want to wait for the VPN connection to be up and configure the connected containers to check, you can either get the JSON from:

- GET `/v1/openvpn/status` which gives `{"status": "running"}` once the VPN has been running for 100ms without error (and the firewall is of course up at this point). Otherwise it would be something else such as `{"status": "starting"}`.
- GET `/v1/publicip/ip` which gives `{"public_ip": "something", "otherfields": "..."}`. If the connection is not up, the `public_ip` value is empty (`""`).

These routes should all return `200` status codes, unless there is some dramatical bug somewhere where it would be a `500`. You can also use `jq` to parse the JSON if you want to.

**Reply by yesimanoob** [reply]
Thank you for confirming what ` {"status": "running"} ` means as well as the background information on the firewall launch timing.

> I'm working now on decreasing that 60ms to 6ms, I'll let you know once I'm done.

That would be great to know, for peace of mind at least. Thank you! I look forward to the relevant note in a future changelog. :)

**Reply by qdm12** [reply]
Please subscribe to issue #695 for that faster firewall setup. I am still working on a rather large configuration reading/parsing refactoring which is currently blocking it, but it will ultimately be done.

**Reply by iluvatyr** [reply]
Not sure if im really blind, but when I go to the wiki, I cant see this page about the Control server in the navigation options on the right that leads to the page linked above HTTP Control Server. I can only find HTTP control server options through the navigation there. Is it me who is blind or is it missing there?

**Reply by qdm12** [reply]
Indeed it was missing, I just added it. Here it is as well: https://github.com/qdm12/gluetun/wiki/HTTP-control-server

Also I improved the firewall setup which now takes 5/10ms to be done vs the previous 75ms.
